tests/doc.rs

```diff
@@ -612,3 +613,97 @@ fn plugins_no_use_target() {
                  .arg("-v"),
                 execs().with_status(0));
 }
+
+#[test]
+fn doc_all_workspace() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [project]
+            name = "foo"
+            version = "0.1.0"
+
+            [dependencies]
+            bar = { path = "bar" }
+
+            [workspace]
+        "#)
+        .file("src/main.rs", r#"
+            fn main() {}
+        "#)
+        .file("bar/Cargo.toml", r#"
+            [project]
+            name = "bar"
+            version = "0.1.0"
+        "#)
+        .file("bar/src/lib.rs", r#"
+            pub fn bar() {}
+        "#);
+    p.build();
+
+    // The order in which bar is compiled or documented is not deterministic
+    assert_that(p.cargo_process("doc")
+                 .arg("--all"),
+                execs().with_stderr_contains("[..] Documenting bar v0.1.0 ([..])")
+                       .with_stderr_contains("[..] Compiling bar v0.1.0 ([..])")
+                       .with_stderr_contains("[..] Documenting foo v0.1.0 ([..])"));
```

Comment:
Could you add `.with_status(0)` assertions to these as well to ensure we test that the process ran successfully?